plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetTypeTranslator.java

```diff
@@ -79,6 +82,10 @@ private ParquetTypeTranslator() {}
             if (fromParquetType == INT96) {
                 return Optional.of(new LongTimestampToVarcharCoercer(TIMESTAMP_NANOS, varcharType));
             }
+            // Binary data type doesn't have any annotation
+            if ((fromParquetType == BINARY || fromParquetType == FIXED_LEN_BYTE_ARRAY) && typeAnnotation == null) {
+                return Optional.of(createVarbinaryToVarcharCoercer(varcharType, false));
+            }
```

Comment:
Are you sure this is needed ? My impression is that this should already work in parquet reader

Reply:
But when reading the slice do we fix/replace the invalid utf-8 bytes ?

Reply:
We don't do that, but the parquet reader actually allows reading this combination. Is this replacing of invalid utf8 bytes something specific to hive ?

Reply:
This replacing on invalid utf8 is a part of specific to hive

Reply:
Okay, can you check if this code will be reached only for hive connector ?
I get the impression that we use the same page source factory for hive, hudi and delta.

Reply:
In case of Delta we won't be facing this issue as it has a strict evaluation checks but for hudi not sure. We need test coverage to validate the behaviour - I was thinking on skipping this coercion for un-partitioned parquet tables, work on schema evolution tests for hudi and then implement it for parquet as well. WDYT ?

Reply:
ya, we can keep parquet for a follow-up change